### PR TITLE
[ci] use composer stage

### PR DESCRIPTION
**Pull request by v1v**
## What

Use composer as part of the CI. For such, I simplified the CI pipeline in favour of a Makefile and added composer to the `Dockerfile`

## Why

Composer build was not available in the CI.

### CODE REVIEW

File: .ci/Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+.PHONY: prepare
+prepare:
+	docker build \
+		--build-arg PHP_VERSION=${PHP_VERSION} \
+		--tag $(IMAGE):${PHP_VERSION} .
+
+.PHONY: build
+build: prepare
+	# docker as the current user
+	docker run --rm -t \
```
Comment (v1v) [inline .ci/Makefile:16]: I could refactor the `docker run` steps just a bit to be reused but there are three different scenarios:
- build/test uses the same commands
- install requires sudo access
- composer requires further `CMD` and sudo access

File: .ci/Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	docker run -t --rm \
+		-v $(PWD):/app \
+		$(IMAGE):${PHP_VERSION} \
+		sh -c 'make install \
+			&& echo 'extension=elastic_apm.so' > /usr/local/etc/php/php.ini \
+			&& echo 'elastic_apm.bootstrap_php_part_file=/app/src/bootstrap_php_part.php' >> /usr/local/etc/php/php.ini \
+			&& php -m \
+			&& cd /app \
+			&& composer install \
+			&& composer static_check_and_run_component_tests_standalone_envs'
```
Comment (v1v) [inline .ci/Makefile:50]: Is this the composer command we wanna use?

File: Dockerfile [hunk 3]
```diff
@@ -7,9 +7,15 @@ RUN apt-get -qq update \
     autoconf \
     curl \
     libcurl4-openssl-dev \
+    procps \
 	--no-install-recommends \
  && rm -rf /var/lib/apt/lists/*
 
+RUN php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" \
+ && php -r "if (hash_file('sha384', 'composer-setup.php') === 'e5325b19b381bfd88ce90a5ddb7823406b2a38cff6bb704b0acc289a09c8128d4a8ce2bbafcd1fcbdc38666422fe2806') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;" \
```
Comment (mdelapenya) [inline Dockerfile:15]: Maybe we can define the hash as an ENV to have it located above in the Dockerfile.

As a follow up, not sure if it's possible to use different installers/different hashes. Then an ARG could be possible to parametrise the build

Comment (v1v) [reply Dockerfile:15]: Sounds good to me, in any case I didn't do any magic but using the installation steps in https://getcomposer.org/download/

**Pull request merged**